File: 4. Java/EX52.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX52
{
    public static void main(String[] args) 
    {
        int qtd, qtdF = 0, qtdM = 0;
        double mai = 0, men = 0, acuAltM = 0;
        char sex;

        Locale.setDefault(Locale.US);
       
        try (Scanner ent = new Scanner(System.in)) 
        {
            System.out.print("# Quantas pessoas serao digitadas: ");
            qtd = ent.nextInt();

            double[] vetAlt = new double[qtd];
            
            for (int i = 0; i < qtd; i++)
            {
                System.out.printf("\n# Altura da %da pessoa: ", i + 1);                
                vetAlt[i] = ent.nextDouble();
                
                // Inicializa maior e menor
                if (i == 0)
                {
                    mai = vetAlt[i];
                    men = vetAlt[i];
                }
                
                // Atualiza maior e menor
                if (mai < vetAlt[i]) mai = vetAlt[i];
                if (men > vetAlt[i]) men = vetAlt[i];
                
                System.out.printf("# Genero da %da pessoa: ", i + 1);
                sex = ent.next().toUpperCase().charAt(0);
                
                // Conta alturas de mulheres e homens
                if (sex == 'F')
                {
                    acuAltM += vetAlt[i];
                    qtdF++;
                }
                else
                {
                    qtdM++;
                }
            }
            
            // Exibe resultados
            System.out.printf("\n# Menor altura = %.2f", men);
            System.out.printf("\n# Maior altura = %.2f", mai);
            
            if (qtdF > 0)
            {
                System.out.printf("\n# Media das alturas das mulheres = %.2f", (acuAltM / qtdF));
            }
            
            else
            {
                System.out.println("# Nao ha mulheres para calcular a media");
            }
            
            System.out.printf("\n# Numero de homens = %d\n", qtdM);
        }
    }
}

File: 6. Python/EX52.py
# Inicializa variáveis para maior e menor altura, quantidade de homens e mulheres e acumulação de altura de mulheres
mai = float('-inf')  # Inicia com o menor valor possível
men = float('inf')   # Inicia com o maior valor possível
qtdF = 0
qtdM = 0
acuAltM = 0
listaGen = ['F', 'M']

# Solicita a quantidade de pessoas a serem digitadas
qtd = int(input("# Quantas pessoas serao digitadas: "))

# Cria vetor para armazenar as alturas
vetAlt = [0.0 for i in range(qtd)]

# Coleta dados das pessoas
for i in range(qtd):
    vetAlt[i] = float(input(f"\n# Altura da {i+1}a pessoa: "))  # Coleta a altura da pessoa

    if vetAlt[i] > mai:  # Atualiza a maior altura se necessário
        mai = vetAlt[i]

    if vetAlt[i] < men:  # Atualiza a menor altura se necessário
        men = vetAlt[i]

    # Coleta o gênero e contabiliza alturas
    while True:
        sex = input(f"# Genero da {i+1}a pessoa (F/M): ")[0].upper()

        if sex in listaGen:  # Verifica se o gênero é válido
            break

        print("# Genero invalido. Por favor, digite 'F' ou 'M'.")

    if sex == 'F':              # Se a pessoa for mulher
        acuAltM += vetAlt[i]    # Acumula a altura das mulheres
        qtdF += 1               # Contabiliza uma mulher
    else:
        qtdM += 1               # Contabiliza um homem

# Exibe menor altura encontrada
print(f"\n# Menor altura = {men:.2f}")

# Exibe maior altura encontrada
print(f"# Maior altura = {mai:.2f}")

# Verifica se há mulheres para calcular a média
if qtdF > 0:
    print(f"# Media das alturas das mulheres = {(acuAltM / qtdF):.2f}")  # Exibe média das alturas das mulheres
else:
    print("# Nao ha mulheres para calcular a media")                     # Mensagem caso não haja mulheres

# Exibe o número de homens
print(f"# Numero de homens = {qtdM}")
